File: frontend/src/app/common/purchase.ts
import { Address } from "./address";
import { Customer } from "./customer";
import { Order } from "./order";
import { OrderItem } from "./order-item";

export class Purchase {
  public customer: Customer = new Customer();
  public shippingAddress: Address = new Address();
  public billingAddress: Address = new Address();
  public order: Order = new Order();
  public orderItems: OrderItem[] = [];

  constructor() {}
}

File: backend/src/main/java/com/wlady/backend/dto/Purchase.java
package com.wlady.backend.dto;

import com.wlady.backend.entity.Address;
import com.wlady.backend.entity.Customer;
import com.wlady.backend.entity.Order;
import com.wlady.backend.entity.OrderItem;
import lombok.Data;

import java.util.Set;

@Data
public class Purchase {
    private Customer customer;
    private Address shippingAddress;
    private Address billingAddress;
    private Order order;
    private Set<OrderItem> orderItems;
}
